src/lib/y2partitioner/widgets/summary_text.rb

```diff
@@ -94,6 +95,9 @@ def refresh_value
       def calculate_actions
         actiongraph = current_graph.actiongraph
         @actions = Y2Storage::ActionsPresenter.new(actiongraph)
+        Y2Storage::DumpManager.dump(current_graph, "partitioner")
+        Y2Storage::DumpManager.dump(@actions)
+        @actions
```

Comment:
I am not sure why the dump is performed here. IMHO, the strategic places would be:

* in storage_manage.rb:
  * after probing
  * after committing
* in inst_disk_proposal.rb
  * after initial proposal
  * after each guided setup
  * after each partitioner
* in partitions_proposal.rb
  * after each partitioner

And maybe, all that places would be included with:

* in storage_manage.rb:
  * after probing
  * after assigning staging=
  * after committing

Reply:
Even more, should not be enough with saving the probed and committed devicegraphs (and actions)?  Well, I guess the idea is to have all steps, even when the changes are not committed.

Reply:
Right; I wanted to be able to reproduce what happened after each step. I had this many times in bug reports about storage-old: Users do a lot of actions; they switch proposal settings back and forth and enter the expert partitioner and whatnot, and at some point they claim "it doesn't work as expected". It's very hard to figure out what the hell they did in all those steps, and in certain cases this turned out to be important.

I believe that storage-ng is a lot more robust and consistent against the user changing settings back and forth, but I'd rather be safe than sorry. And even from the sequence of those files you can instantly tell what a user did: Just accept one proposal, changed only one or two proposal settings (-> xx-proposal.yml), entered the partitioner (-> xx-partitioner.yml), maybe abandoned his changes and resort to the proposal after all (-> xx-proposal.yml), and, most importantly, what was actually committed.

As for why do it in _calculate_actions_: Because the actions are already calculated here, they are dumped; and to be very sure which version of the staging devicegraph it refers to, simply dump it one more time. They should even have the same timestamp (it's only seconds resolution). I wanted to avoid creating yet another actiongraph and ActionsPresenter, and, more importantly, I want to log exactly what we showed the user and not just an approximation. 

Call me paranoid; just because I am paranoid that doesn't mean that they are **not** out there to get me. ;-)

Reply:
Moreover, when the Partitioner is used in a running system, a last step with the summary of actions is shown, so also in this case the files will be saved. Probably, this dump could be moved to `Dialogs::Main#run`, so they are generated only after accepting the Partitioner. What do you think?

Reply:
Not having reviewed the whole PR (I just got attracted by the mails about this particular point), I also believe it's overkill to save the actions every time the user visualize them. That means:

 - every time the user clicks on the "installation summary" section of the partitioner (something I do regularly while using it)
 - every time the user visits the proposal summary screen that is displayed always in AutoYaST and some times even in normal installation of some products
 - every time the user passes by the corresponding installation screen, no matter if going back, going forward, having accepted or CANCELED the guided setup or the proposal settings
 - on the confirmation step of the partitioner when running on an installed system
 - surely more cases

Logging the actions way more often than logging the devicegraphs can be very confusing (not to say harmful for rotation). And to add more confusion, it implies logging partial results (like when the user clicks on the summary in the middle of a Partitioner execution) and/or discarded results (when the user doesn't accept a result). Being paranoid can hurt yourself. ;-)

In my opinion, it only makes sense to log the actions at the same point that we log devicegraphs. A new set of actions is only the consequence of a new devicegraph, logging partial results or logging the same set of actions over and over only gives the false impression about something happening in points in which nothing is really moving.

In general, I agree that this should be more than enough:

- During probing and sanitizing of the result
- When assigning `StorageManager#staging`
- During commit

In all the three cases we could log both the devicegraph and its associated actiongraph.

That should cover all devicegraphs generated by the partitioner or by the guided setup. And should exclude partial or discarded results (that are noise in 99% of the cases).

Reply:
In the running system you might get one more, depending where the user clicks. The last step just before commit (which is different in the installed system) always dumps both the devicegraph and the actions; but this is specific to the partitioner in the installed system.

When the user clicks on the "summary" page (which he might or might not do), both graphs are dumped once more. So, in that case (user clicks on summary AND he is presented with the actions just prior to commit) we might get two identical files of each kind, but in what way should that be a problem?

Reply:
Well, so we might get some extra ones. This is not a problem with rotation at all since it's not individual files that are rotated, but the whole directory that contains those files - for each program call (as mentioned in the initial description of this PR).

In the course of the last 3 years, I had to debug countless problems where I had to fiddle that stuff out of all that completely irrelevant noise in the y2logs (more often than not not even knowing what program run was relevant since the y2logs accumulate all of them in the installed system). It turned out to be necessary to have all that information; some users do a lot of actions to get to the final result, and it is absolutely vital to be able to reproduce what they did.

Those dump files will only accumulate if a user does a lot of actions: Change proposal settings and regenerate the proposal, enter the expert partitioner and whatnot. If a user only accepts the proposal, it will only be 3 sets:

- probed
- proposal
- committed

That's it.

When the user keeps clicking around, there is no way to find out what happened without those dump files; not even the macro recorded during installation is any useful since you never know exactly what he clicked in the tree or in the table and what happened then. This is why I want (and I **insist**!) in having this detailed information.

We will have to maintain that stuff for about 10 years. If we don't add this now, we will have to live with bug reports coming in during all those 10 years where we will have no way of knowing what happened.

So, if it turns out to be too many of those dump files, we can simply ignore them, just as we ignore 99.8% of the y2logs (just look at all the garbage in there). But the one information we don't record might be the one we sorely miss when a user (typically a paying L3 customer) has a problem.

So, without even having seen any of the results of this, why would we throw away information that we can easily keep -- and that will be essential in the future?

As of now, not even the actions are recorded in the y2log; I wonder how anybody wants to reproduce or track down any bug reports coming in like this. I don't believe in miracles; I don't believe in bugs mysteriously debugging and fixing themselves.

I want those dump files. And I want as many of them as I can get.

It's easy to simply ignore some of them. But it's impossible to retrieve them when they are not recorded.

Reply:
Unlike the y2logs, you can simply discard the ones you don't want or need. The y2logs are just sequential text files where you'd have to write your own tools to filter out all the noise (and in the past, most of the team did exactly that).

If you prefer to see only the final result, well, fine, then just look at `*probed.*` and  `*commited.*`. That's it (and that's also why they have special names that are easy to tell apart from the rest). For everything else, we record more of them which might or might not be relevant for the bug you are currently working on. And that's also why there is also a line in the y2log (which was snwint's idea) _"Dumping devicegraph staging to 03-proposal"_ so you can easily find the corresponding place in the y2logs (or the other way round).

As I wrote above, if it turns out to be too much, we can easily throw out some of those places later - for SLE-15 SP1, for example. But please let's get some hands-on experience with it first so we can get an idea how useful it is.

Since the y2logs have degenerated to the point of uselessness with all the clutter (but omitting vital things, mostly for the sake of brevity in the Ruby code, I guess), we desperately need useful debugging tools. And this is one of it.

Reply:
Ok. Fine for me. Let's go for it if you feel they might be useful.

My main concerns right now are log rotation on an installed system and memory consumption during installation (sadly we are in the point in which the approximately 40KiB used by every dump[1] makes a difference).

And just for the record, I have never missed information with the current level of logging while debugging partitioner or proposal bugs (and we got some reports for SLE15 and Leap15 already). I appreciate the fact of having the XML and yaml in separate files, since it's indeed more convenient than copying&pasting the XML from y2logs and then using the Ruby interpreter to make sense of the dump. But that's more about convenience than about missing information. So far, debugging what happened exactly has always[2] been possible.

[1] Rough estimation for a in a system with subvolumes, based on the files we have right now in `test/data`

[2] "always" means the last 5 months or so. The first versions of storage-ng really lacked proper logging, but that's pretty much fixed now.